impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -315,7 +331,7 @@ public synchronized void removeBroker(int nodeId) throws IllegalArgumentExceptio
             throw new IllegalStateException("Cannot remove nodes from a cluster with stopped nodes.");
         }
 
-        var target = servers.keySet().stream().filter(n -> n != nodeId).findFirst();
+        var target = servers.keySet().stream().filter(n -> n != nodeId && clusterConfig.hasBrokerRole(n)).findFirst();
```

Comment:
is `&& clusterConfig.hasBrokerRole(n)` really needed in the filter clause? Shouldn't we have validated that up front that `clusterConfig.hasBrokerRole(nodeId)` is true?

Reply:
This is about finding the target brokers for replicas currently on `nodeId`, I.e. `n` here is not the same as `nodeId`, and without the extra condition `n` might be a controller, but not a broker, thus cannot replicate any partitions hosted the broker which is being removed.